src/views/templates/actions/components/CloneTemplateModal.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+
+import { TemplateModel, V1Template } from '@kubevirt-ui/kubevirt-api/console';
+import TabModal from '@kubevirt-utils/components/TabModal/TabModal';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { k8sCreate, K8sResourceCommon } from '@openshift-console/dynamic-plugin-sdk';
+import { ButtonVariant, Form, FormGroup, TextInput } from '@patternfly/react-core';
+
+type CloneTemplateModalProps = {
+  isOpen: boolean;
+  obj: V1Template;
+  onClose: () => void;
+};
+
+const CloneTemplateModal: React.FC<CloneTemplateModalProps> = ({ isOpen, obj, onClose }) => {
+  const { t } = useKubevirtTranslation();
+  const [templateName, setTemplateName] = React.useState('');
+  const [templateProvider, setTemplateProvider] = React.useState('');
+
+  const onSubmit = async () => {
+    await k8sCreate<V1Template>({
+      model: TemplateModel,
+      data: {
+        ...obj,
+        metadata: {
+          annotations: {
+            ...obj?.metadata?.annotations,
+            'template.kubevirt.io/provider': templateProvider,
+          },
+          labels: { ...obj?.metadata?.labels, 'template.kubevirt.io/type': 'vm' },
+          name: templateName,
+          namespace: obj?.metadata?.namespace,
+        },
+      },
+    });
+  };
```

Comment:
```suggestion
const cloneTemplate = rfc6902.applyPatch(obj, [
  {op: 'replace', path: '/metadata/labels/template.kubevirt.io/type', value: 'vm'},
  {
      op: 'replace',
      path: '/metadata/annotations/template.kubevirt.io/provider',
      value: templateProvider
  },
  ]);

  const onSubmit = async () => {
    await k8sCreate<V1Template>({
      model: TemplateModel,
      data: cloneTemplate,
    });
  };
```

Reply:
I am not sure if we really need to add a new package to our project just because of this...

Reply:
@hstastna hi, you don't need to add a new package, see:
https://github.com/kubevirt-ui/kubevirt-plugin/pull/199#discussion_r850073925

> to change only one element in an obj we have immer you can look how it's used in the wizard
> another option is to use some rfc6902 package that allow to patch an object, but I don't think we have one

in the code snippet here you use the spread operator to mutate the template object. using the spread operator is a good option to do light mutations, but it becomes less appealing when the object being mutated is deep, or you have more then a few changes you want to do in the object.

In https://github.com/kubevirt-ui/kubevirt-plugin/pull/199#discussion_r850073925 I suggested `immer` or some rfc6902 complaint library not necessary `rfc6902`, any library that conform to the rfc6902 standard will do.

spread operator:
pros:
light and built into the language, no need to import anything, easy to read and maintain on shallow object. 
cons:
we have to carefully shallow copy every level of the state structure, make is hard to understand and maintain on deep objects.

Immer:
pros:
works nice with typescript and react
cons:
require external library

using rfc6902 standard:
pros:
using a standard, works well with k8s and openshift that use this standard
cons:
require external library

if you wish to use a helper library instead of using the spread operator as suggested in https://github.com/kubevirt-ui/kubevirt-plugin/pull/199#discussion_r850073925, you can try other options like `immer` or implement an rfc6902 standard complient helper or try some other solution, you do not need to add a new package.

Reply:
Thank you very much, I will reconsider pros and cons you've mentioned above and will update the PR.